rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceFilterRequest.java

```diff
@@ -0,0 +1,146 @@
+    public static final String COUNT_KEY = "count";
+
+    private String fromBlock;
+    private String toBlock;
+    private List<String> fromAddress;
+    private List<String> toAddress;
+    private Integer after;
+    private Integer count;
+
+    public static TraceFilterRequest buildFrom(Map<String, Object> map) {
+        TraceFilterRequest request = new TraceFilterRequest();
+
+        if (map.containsKey(FROM_BLOCK_KEY)) {
+            request.setFromBlock(map.get(FROM_BLOCK_KEY).toString());
+        }
+
+        if (map.containsKey(TO_BLOCK_KEY)) {
+            request.setToBlock(map.get(TO_BLOCK_KEY).toString());
+        }
+
+        if (map.containsKey(FROM_ADDRESS_KEY)) {
+            request.setFromAddress((List<String>) map.get(FROM_ADDRESS_KEY));
+        }
+
+        if (map.containsKey(TO_ADDRESS_KEY)) {
+            request.setToAddress((List<String>) map.get(TO_ADDRESS_KEY));
+        }
+
+        if (map.containsKey(AFTER_KEY)) {
+            request.setAfter(Integer.valueOf(map.get(AFTER_KEY).toString()));
+        }
+
+        if (map.containsKey(COUNT_KEY)) {
+            request.setCount(Integer.valueOf(map.get(COUNT_KEY).toString()));
+        }
+
+        return request;
+    }
+
+
+    public BigInteger getFromBlockNumber() {
+        return HexUtils.stringHexToBigInteger(this.fromBlock);
+    }
+
+    public String getFromBlock() {
+        return fromBlock;
+    }
+
+    public void setFromBlock(String fromBlock) {
+        this.fromBlock = fromBlock;
+    }
+
+    public BigInteger getToBlockNumber() {
+        return HexUtils.stringHexToBigInteger(this.toBlock);
+    }
+
+    public String getToBlock() {
+        return toBlock;
+    }
+
+    public void setToBlock(String toBlock) {
+        this.toBlock = toBlock;
+    }
+
+    public List<BigInteger> getFromAddressAsBigIntegers() {
+        if (this.fromAddress == null) {
+            return new ArrayList<>();
+        }
+
+        return this.fromAddress.stream().map(HexUtils::stringHexToBigInteger).collect(Collectors.toList());
+    }
+
+    public List<String> getFromAddress() {
+        return this.fromAddress;
+    }
+
+    public void setFromAddress(List<String> fromAddress) {
+        this.fromAddress = fromAddress;
+    }
+
+    public List<String> getToAddress() {
+        return toAddress;
+    }
+
+    public void setToAddress(List<String> toAddress) {
+        this.toAddress = toAddress;
+    }
+
+    public List<BigInteger> getToAddressAsBigIntegers() {
+        if (this.toAddress == null) {
+            return new ArrayList<>();
+        }
+
+        return this.toAddress.stream().map(HexUtils::stringHexToBigInteger).collect(Collectors.toList());
+    }
+
+    public Integer getAfter() {
+        return after;
+    }
+
+    public void setAfter(Integer after) {
+        this.after = after;
+    }
+
+    public Integer getCount() {
+        return count;
+    }
+
+    public void setCount(Integer count) {
```

Comment:
Do we really need this kind of setters? We can access the fields directly.

If you want to keep them, do we really need them to be `public`? I see no access from the "outside" and I'd rather keep them private to avoid undesired external changes and promote immutability if the only way to instantiate it is via `buildFrom`.

WDYT?